Diffusion/Common/State.swift

```diff
@@ -126,4 +126,25 @@ class Settings {
             return ComputeUnits(rawValue: current)
         }
     }
+    
+    public func applicationSupportURL() -> URL {
+        let fileManager = FileManager.default
+        guard let appSupportURL = fileManager.urls(for: .applicationSupportDirectory, in: .userDomainMask).first else {
+            // To ensure we don't return an optional - if the user domain application support cannot be accessed use the top level application support directory
+            return URL.applicationSupportDirectory
+        }
+
+        let appBundleIdentifier = Bundle.main.bundleIdentifier ?? ""
+        let appDirectoryURL = appSupportURL.appendingPathComponent(appBundleIdentifier)
```

Comment:
I wouldn't add this piece to maximize compatibility, especially for people that have already downloaded the app from the Mac App Store. When we make a new release, it'd be nice if they don't have to download all models again (and have an old folder with orphaned models).

Reply:
How about changing the nil check from a blank string to the default bundle id?   let appBundleIdentifier = Bundle.main.bundleIdentifier ?? "com.huggingface.diffusers"

Reply:
No, I mean I wouldn't add the bundle identifier because that way it's compatible with the existing location (at least for the app released in the App Store). The current location is already inside a container exclusive to the app, so we don't need to disambiguate it further. For debug apps, it would have been better to use this location you propose from the start, but I think it's important to keep compatibility if we can. Do you think it could be a problem down the line?

Reply:
Thank you for clarifying that.

Reply:
The only thing I see as an issue is when we start adding additional support folders. In other PRs we introduce a temp storage url -  hf-diffusion-tmp. If under a single bundle ID then all of our "top level" support content is easy to find. When I was working on drag and drop I was placing .png temporary files into the same level as the hf-diffusion-models support folder which on local  development testing was littering the application support folder. 

I don't think it's a massive problem  to have a few support files for the same app at that level, especially since local dev runs are the ones that would be affected.

Reply:
Yeah, I should have been more thoughtful when I started, I didn't think about potential temp files then. Thanks for the explanation!